src/figuresClasses.ts

```diff
@@ -1,19 +1,108 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public color: Color;
+
+  public a: number;
+
+  public b: number;
+
+  public c: number;
+
+  constructor(
+    color: Color,
+    a: number,
+    b: number,
+    c: number,
+  ) {
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
 
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error(`Wrong sides - a: ${a}, b: ${b}, c: ${c}!!!`);
+    }
+
+    if ((a + b) <= c || (b + c) <= a || (c + a) <= b) {
+      throw new Error('Incorrect proportion of the sides of a triangle!');
+    }
+  }
+
+  shape: Shape = 'triangle';
+
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
+    const S = Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c));
+    const normalizeArea: string = S.toFixed(3);
```

Comment:
you make this normalize 3 times
So, make a separate function for this & reuse in all cases

Reply:
done, thx for advice)))